client/src/Components/RelatedItemsAndComparison/RelatedCards.jsx

```diff
@@ -3,33 +3,61 @@ import axios from 'axios';
 import ReactDOM from 'react-dom';
 import ComparisonModal from './ComparisonModal.jsx'
 import styled from 'styled-components';
+import StarRating from '../SharedComponents/StarRating.jsx';
 
 const RelatedCards = (props) => {
-  const [modal, setModal] = useState(false)
-  const [products, setProducts] = useState([])
+  const [modal, setModal] = useState(false);
+  const [products, setProducts] = useState([]);
+  const [clickedProd, setClickedProd] = useState({});
   var modalToggle = () => {
     setModal(!modal)
   }
+
   useEffect(() => {
-    console.log('index', props.relatedIndex)
-    console.log('slice', props.relatedProd.slice(props.relatedIndex, props.relatedIndex + 4))
-    console.log('related id', props.relatedId)
     setProducts(props.relatedProd.slice(props.relatedIndex, props.relatedIndex + 4));
   }, [props.relatedIndex, props.relatedProd])
+
   async function handleCardClick(id) {
     return props.setProductId(id).then(() => {props.setRelatedIndex(0)})
   }
+
+  const getUrl = (id) => {
+    if (props.styles !== []) {
+      for (var i = 0; i < props.styles.length; i++) {
+        if (props.styles[i].product_id == id) {
+          return props.styles[i].results[0].photos[0].thumbnail_url
+        }
+      }
+    }
+  }
+  const getRatings = (id) => {
+    if (props.reviewData !== []) {
+      for (var i = 0; i < props.reviewData.length; i++) {
+        if (props.reviewData[i].product_id == id) {
+          return props.reviewData[i].ratings
+        }
+      }
+    }
+  }
   return (
     <>
     {products?.map((prod) => {
+      {console.log('url', getUrl(prod.id))}
       return (
           <CardBox key={prod.id}>
-            <div onClick = {modalToggle}>⭐️</div>
-            {modal ? <ComparisonModal modalToggle = {modalToggle} clickedProd={prod} chosenProduct={props.chosenProduct}/> : null }
-            <p onClick={(e) => {handleCardClick(prod.id)}}>{prod.category}</p>
-            <p onClick={(e) => {handleCardClick(prod.id)}}>{prod.name}</p>
-            <p onClick={(e) => {handleCardClick(prod.id)}}>{prod.default_price}</p>
-            <p onClick={(e) => {handleCardClick(prod.id)}}>Star Rating</p>
+            <div onClick={(e) => {
+              setClickedProd(prod);
+              modalToggle();}}>⭐️</div>
+              {modal ? <ComparisonModal modalToggle = {modalToggle} clickedProd={clickedProd} chosenProduct={props.chosenProduct}/> : null }
+              <div onClick={(e) => {handleCardClick(prod.id)}}>
+              <ThumbnailImage src={getUrl(prod.id)} />
+              <p>{prod.category}</p>
+              <p>{prod.name}</p>
+              <p>{prod.default_price}</p>
+              <StarContainer>
+                <StarRating reviewData={getRatings(prod.id)}/>
+              </StarContainer>
+            </div>
```

Comment:
Your modal here is implemented nicely. Note that when the modal pops up, we can still see the filled stars in all of the shared star rating components. I presume it's due to the z-index setting of said stars. Nothing serious here, just a small styling thing you may want to look at (feel free to play around with the z-index of the said star rating shared component if need be).